open-labz/react-javascript-starter
# Initialise the Webpack environment

#### What does this PR do?
This PR implements the Webpack environment required to bundle the application's code and assets...whereby the bundled code/assets are now ready to be distributed on the server as a single package, instead of multiple files. Webpack also works with Babel to transpile ES6 code during the bundling process.

We use the following toolbox: 
- [Webpack](https://webpack.js.org)
- [Babel](https://babeljs.io)
- [Webpack Plugins](https://webpack.js.org/plugins/html-webpack-plugin)

#### Description of Task to be completed?
- install Webpack dependencies.
- create sample app starter files.
- install Babel transpiler dependencies.
- configure Babel settings.
- configure Webpack settings.
- configure HTML webpack plugin.

#### How should this be manually tested?

> Once you have cloned and setup the project repo locally, and installed dependencies with `npm install`...
 
- In your terminal, browse to the project directory.
- Run the npm script `npm run start:dev` to trigger the build process.
- Check for bundled files in the created `dist/` directory.

#### Any background context you want to provide?
- **ES6** - This is a recent major update to JavaScript that includes dozens of new features. Many browsers may not support it currently, which is why we need a transpiler like Babel.

- **Webpack plugins** - webpack has a rich plugin interface. Most of the features within webpack itself use this plugin interface. This makes webpack flexible, allowing you to extend its functionality.

#### What are the relevant pivotal tracker stories?
N/A

#### Screenshots

##### Files Bundled by Webpack within the project
![Screenshot - Project dist/ folder](https://user-images.githubusercontent.com/65714243/91484394-6fbd1080-e8a9-11ea-9413-c94f1e72391b.png)
